packages/metrics/src/features/metrics/utils.ts

```diff
@@ -109,6 +109,38 @@ export const generateEmptyBirthKeyFigure = (
   }
 }
 
+export function createEstimatesArray(
+  years: number[],
+  output: number[],
+  extension: fhir.Extension
+) {
+  const locationExtensions = JSON.parse(extension.valueString as string) as {
+    [key: string]: number
+  }[]
+  // Imagine that years searching: [2022,2023] & extensions is: [{"2015": 17.2}, {"2022": 16}, {"2017": 17.2}]
+  // We have no data for 2023 so we want to use 2022 estimates
+
+  for (let i = 0; i < years.length; i++) {
+    /*
+   This code calculates the difference between each year in the array and the target year 2023, then 2022,
+   and then selects the item with the smallest difference.
+   The reduce function is used to iterate over each item in the array and compare the differences.
+   Finally, it returns the item with the closest year to 2023, then 2022.
+  */
+    const closestItem = locationExtensions.reduce((prev, curr) => {
+      const key = parseInt(Object.keys(curr)[0])
+      const prevKey = prev ? parseInt(Object.keys(prev)[0]) : Infinity
+      const currDiff = Math.abs(key - years[i])
+      const prevDiff = Math.abs(prevKey - years[i])
+      return currDiff < prevDiff ? curr : prev
+    }, null)
+    const closestValue = closestItem ? Object.values(closestItem)[0] : null
+    if (closestValue) {
+      output.push(closestValue)
+    }
+  }
+}
```

Comment:
Sorry, didn't notice this before and the calling function is of course what it is but you could turn this into a pure function with minor adjustments

```suggestion
export function createEstimatesArray(
  years: number[],
  extension: fhir.Extension
) {
  const output: number[] = []
  const locationExtensions = JSON.parse(extension.valueString as string) as {
    [key: string]: number
  }[]
  // Imagine that years searching: [2022,2023] & extensions is: [{"2015": 17.2}, {"2022": 16}, {"2017": 17.2}]
  // We have no data for 2023 so we want to use 2022 estimates

  for (let i = 0; i < years.length; i++) {
    /*
   This code calculates the difference between each year in the array and the target year 2023, then 2022,
   and then selects the item with the smallest difference.
   The reduce function is used to iterate over each item in the array and compare the differences.
   Finally, it returns the item with the closest year to 2023, then 2022.
  */
    const closestItem = locationExtensions.reduce((prev, curr) => {
      const key = parseInt(Object.keys(curr)[0])
      const prevKey = prev ? parseInt(Object.keys(prev)[0]) : Infinity
      const currDiff = Math.abs(key - years[i])
      const prevDiff = Math.abs(prevKey - years[i])
      return currDiff < prevDiff ? curr : prev
    }, null)
    const closestValue = closestItem ? Object.values(closestItem)[0] : null
    if (closestValue) {
      output.push(closestValue)
    }
  }
  return output
}
```